src/Grpc.Net.Client/Internal/GrpcProtocolConstants.cs

```diff
@@ -103,15 +115,29 @@ static GrpcProtocolConstants()
             // but in case it isn't then don't include version in user-agent.
             if (assemblyVersion != null)
             {
-                userAgent += "/" + assemblyVersion.Version;
+                // 1.24.0
+                userAgent += $"/{assemblyVersion.Version} ";
             }
 
-            UserAgentHeader = "User-Agent";
-            UserAgentHeaderValue = userAgent;
-            TEHeader = "TE";
-            TEHeaderValue = "trailers";
+            // (.NET 5.0.7;
+            userAgent += $"({RuntimeInformation.FrameworkDescription}; ";
+            // CLR 5.0.0;
+            userAgent += $"CLR {Environment.Version}; ";
+
+            // .NETCoreApp,Version=v6.0; 
+            var targetFramework = assembly
+                .GetCustomAttributes<TargetFrameworkAttribute>()
+                .FirstOrDefault()?
+                .FrameworkName;
+            if (!string.IsNullOrEmpty(targetFramework))
+            {
+                userAgent += $"{targetFramework}; ";
+            }
 
-            DefaultMessageAcceptEncodingValue = GetMessageAcceptEncoding(DefaultCompressionProviders);
+            // x64)
+            userAgent += Environment.Is64BitProcess ? "x64)" : "x32)";
```

Comment:
nit: Grpc.Core uses x86  for 32-bit (and x64 for AMD64).

Also, just supporting AMD architecture won't be enough.  we definitely need at least arm64 (since that's currently supported by .NET and ARM architecture is becoming increasingly important these days).

Reply:
I think you should use  RuntimeInformation.ProcessArchitecture  (which is readily available on platforms that grpc-dotnet currently supports) https://github.com/grpc/grpc/blob/7c10e1b9fe9b751cffc3e2d2c0f37feb3163fe93/src/csharp/Grpc.Core/Internal/CommonPlatformDetection.cs#L80